src/main/webapp/app/shared/metis/posting-header/post-header/posting-header.component.ts

```diff
@@ -29,22 +41,54 @@ export abstract class PostingHeaderDirective<T extends Posting> implements OnIni
     userAuthorityTooltip: string;
     currentUser?: User;
 
+    // Icons
+    readonly faPencilAlt = faPencilAlt;
+    readonly faCheckSquare = faCheckSquare;
+
     protected metisService: MetisService = inject(MetisService);
     protected accountService: AccountService = inject(AccountService);
 
     /**
-     * on initialization: determines if user is at least tutor in the course and if user is author of posting by invoking the metis service,
+     * on initialization: determines if user is author of posting by invoking the metis service,
      * determines if posting is of today and sets the today flag to be shown in the header of the posting
      * determines icon and tooltip for authority type of the author
      */
     ngOnInit(): void {
         this.accountService
             .getAuthenticationState()
-            .pipe(tap((user: User) => (this.currentUser = user)))
+            .pipe(
+                tap((user: User) => {
+                    this.currentUser = user;
+                    this.setUserProperties();
+                }),
+            )
             .subscribe();
-        this.postingIsOfToday = dayjs().isSame(this.posting.creationDate, 'day');
+        this.postingIsOfToday = dayjs().isSame(this.posting()?.creationDate, 'day');
         this.todayFlag = this.getTodayFlag();
+    }
+
+    private isPost(posting: Posting | AnswerPost | undefined): posting is Post {
+        return posting !== undefined && 'resolved' in posting;
+    }
+
+    get isPostResolved(): boolean {
+        const p = this.posting();
+        return this.isPost(p) && p.resolved === true;
+    }
```

Comment:
Could you solve this via computed(..) / effect(..), instead of a method call ? This might be called quite often from the template (you could check that by inserting a console log)